src/dht/client.rs

```diff
@@ -47,6 +59,19 @@ impl Client {
         }
     }
 
+    /// Main asynchronous function that handles network events, stdin input, and shutdown signals.
+    ///
+    /// # Parameters
+    ///
+    /// - `network_events`: A receiver channel (`Receiver<Event>`) that provides incoming network events for processing.
+    /// - `shutdown_dht_client_receiver`: A broadcast receiver (`broadcast::Receiver<bool>`) that listens for a shutdown signal
+    ///   to gracefully terminate the DHT client.
+    ///
+    /// # Returns
```

Comment:
If it doesn't return anything, this doesn't need to be here - mentioning that it runs until it encounters a shutdown signal is enough :+1: